### PR TITLE
Feature/report post revamped

### PR DESCRIPTION
**Pull request by lifenautjoe**

### CODE REVIEW

File: openbook_auth/models.py [hunk 1]
```diff
@@ -3194,6 +3554,61 @@ def _check_is_not_blocked_with_user_with_id(self, user_id):
...
+
+    def _check_can_report_user(self, user):
+        self._check_has_not_reported_user_with_id(user_id=user.pk)
+        if self.pk == user.pk:
+            raise ValidationError(
+                _('You cannot report yourself...'),
+            )
+
+    def _check_has_not_reported_user_with_id(self, user_id):
+        if self.has_reported_user_with_id(user_id=user_id):
```
Comment (evict) [inline openbook_auth/models.py]: So users can only be reported once? Even though different categories are available, and a user might be violating more than one?

Comment (evict) [reply openbook_auth/models.py]: Same goes for the other instances, communities, posts etc.

File: openbook_auth/models.py [hunk 2]
```diff
@@ -233,6 +249,26 @@ def save(self, *args, **kwargs):
...
+        for post in self.posts.all().iterator():
+            post.soft_delete()
+
+        for community in self.created_communities.all().iterator():
+            community.soft_delete()
+
+        self.is_deleted = True
+        self.save()
+
+    def unsoft_delete(self):
```
[the remark this answers is not in the document]
Comment (lifenautjoe) [reply openbook_auth/models.py]: Talked IRL

File: openbook_posts/models.py [hunk 3]
```diff
@@ -231,29 +273,44 @@ class PostImage(models.Model):
...
 
 
 class PostVideo(models.Model):
     post = models.OneToOneField(Post, on_delete=models.CASCADE, related_name='video')
     video = models.FileField(_('video'), blank=False, null=False, storage=post_image_storage,
                              upload_to=upload_to_post_video_directory)
+    hash = models.CharField(_('hash'), max_length=64, blank=False, null=True)
+
+    @classmethod
+    def create_post_video(cls, video, post_id):
```
[the remark this answers is not in the document]
Comment (lifenautjoe) [reply openbook_posts/models.py]: YE

File: openbook_moderation/tests/views/moderated_object.py [hunk 4]
```diff
@@ -0,0 +1,3600 @@
...
+                                                                                   category_id=report_category.pk)
+
+        url = self._get_url(moderated_object=moderated_object)
+        headers = make_authentication_headers_for_user(non_global_moderator)
+        response = self.client.patch(url, data={
+            'description': new_moderated_object_description,
+            'category_id': new_report_category.pk
+        }, **headers)
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
```
[the remark this answers is not in the document]
Comment (lifenautjoe) [reply openbook_moderation/tests/views/moderated_object.py]: must update description

File: openbook_auth/models.py [hunk 5]
```diff
@@ -3194,6 +3554,61 @@ def _check_is_not_blocked_with_user_with_id(self, user_id):
         if self.is_blocked_with_user_with_id(user_id=user_id):
             raise PermissionDenied(_('This account is blocked.'))
 
+    def _check_can_report_comment_for_post(self, post_comment, post):
+        self._check_has_not_reported_post_comment_with_id(post_comment_id=post_comment.pk)
```
[the remark this answers is not in the document]
Comment (lifenautjoe) [reply openbook_auth/models.py]: It could be inverted yee

File: openbook_auth/models.py (hunk 1 above)
Comment (lifenautjoe) [reply openbook_auth/models.py]: There's a description you can add up ur reportz

File: openbook_moderation/views/moderated_object/views.py [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+        serializer.is_valid(raise_exception=True)
+        data = serializer.validated_data
+
+        count = data.get('count')
+        max_id = data.get('max_id')
+
+        user = request.user
+
+        with transaction.atomic():
+            moderated_object_logs = user.get_logs_for_moderated_object_with_id(count=count, max_id=max_id)
```
[the remark this answers is not in the document]
Comment (lifenautjoe) [reply openbook_moderation/views/moderated_object/views.py]: Added this + tests

File: openbook_moderation/views/moderated_object/views.py [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+            moderated_object = user.unverify_moderated_object_with_id(moderated_object_id=moderated_object_id, )
+
+        moderated_object_serializer = ModeratedObjectSerializer(moderated_object,
+                                                                context={"request": request})
+
+        return Response(moderated_object_serializer.data, status=status.HTTP_200_OK)
+
+
+class ModeratedObjectLogs(APIView):
+    permission_classes = (IsAuthenticated,)
```
[the remark this answers is not in the document]
Comment (lifenautjoe) [reply openbook_moderation/views/moderated_object/views.py]: Adding it

File: openbook_posts/views/post/views.py [hunk 8]
```diff
@@ -133,7 +133,7 @@ def post(self, request, post_uuid):
 
 
 class PostClose(APIView):
-    permission_classes = (IsAuthenticated,)
+    permission_classes = (IsAuthenticated, IsNotSuspended)
```
[the remark this answers is not in the document]
Comment (lifenautjoe) [reply openbook_posts/views/post/views.py]: Yeah theres some 3 of these where we I'll add the is not suspended. In general, it should only be for APIs that get other people's content or create content

**Pull request merged**